server/index.js

```diff
@@ -1,37 +1,309 @@
-'use strict';
+"use strict";
 
-const argv = require('yargs').argv;
+// Server
+const fs = require('fs');
+const path = require('path');
+const express = require('express');
+const helmet = require('helmet');
+const favicon = require('serve-favicon');
+const { argv } = require('yargs');
+const chalk = require('chalk');
 
-if ( argv.production ) {
-  // Force production environment
-  process.env.NODE_ENV = 'production';
+// Lib
+const getDevice = require('./getDevice');
+
+// ------------------------------------------------------------------------------
+// Initialize & Configure Application
+// ------------------------------------------------------------------------------
+
+const PRODUCT = 'lwjgl.org';
+const app = express();
+const config = require('../config.json');
+
+app.set('port', config.server.port);
+app.set('view engine', 'pug');
+app.set('views', __dirname + '/views');
+
+app.enable('case sensitive routing');
+app.enable('strict routing');
+app.disable('x-powered-by');
+
+app.set('trust proxy', [
+  'loopback',
+  'linklocal',
+  'uniquelocal',
+  // Cloudfront subnets
+  '13.32.0.0/15',
+  '52.46.0.0/18',
+  '52.84.0.0/15',
+  '52.222.128.0/17',
+  '54.182.0.0/16',
+  '54.192.0.0/16',
+  '54.230.0.0/16',
+  '54.239.128.0/18',
+  '54.239.192.0/19',
+  '54.240.128.0/18',
+  '204.246.164.0/22',
+  '204.246.168.0/22',
+  '204.246.174.0/23',
+  '204.246.176.0/20',
+  '205.251.192.0/19',
+  '205.251.249.0/24',
+  '205.251.250.0/23',
+  '205.251.252.0/23',
+  '205.251.254.0/24',
+  '216.137.32.0/19'
+]);
+
+app.locals.development = app.get('env') === 'development';
+app.locals.production = !app.locals.development;
+
+// View options
+app.locals.doctype = 'html';
+app.locals.pretty = app.locals.development || argv.pretty ? '  ' : false;
+app.locals.compileDebug = false;
+app.locals.cache = app.locals.production && argv.nocache === undefined;
+
+if ( app.locals.development ) {
+  console.log(chalk.yellow(`Starting ${PRODUCT} in ${app.get('env')} mode`));
 }
 
-require('babel-core/register')({
```

Comment:
It might, although they are talking about performance issues (Babel has to transpile ES6+ code before Node can launch the server). It delays launching the server, not that the code runs slower.

If you pre-compile files to a "server-build" this performance penalty is removed. But since the same code is generated, I doubt it is related to memory leaks.